### PR TITLE
增加对socks5的支持

### PR DESCRIPTION
**Pull request by CoolDarran**

### CODE REVIEW

**Comment by CoolDarran** [comment]
@geekdada 看看

**Comment by geekdada** [comment]
🙆‍♂️

File: docs/guide/custom-provider.md [hunk 1]
```diff
@@ -277,6 +277,25 @@ module.exports = {
...
+{
+  type: 'socks5',
+  nodeName: '🇭🇰HK',
+  hostname: 'hk.example.com',
+  port: 80,
+  username: 'username', // 可选
+  password: 'password', // 可选
+  tls: true, // 可选
+  skipCertVerify: true, // 可选
+  udp: false, // 可选, clash配置
```
Comment (geekdada) [inline docs/guide/custom-provider.md]: 这个 Key 名我想统一叫 `udpRelay`，Vmess 的配置后面也会改过来

File: lib/types.ts [hunk 2]
```diff
@@ -215,6 +216,19 @@ export interface TrojanNodeConfig extends SimpleNodeConfig {
...
 
+export interface Socks5NodeConfig extends SimpleNodeConfig {
+  readonly type: NodeTypeEnum.Socks5;
+  readonly hostname: string;
+  readonly port: number|string;
+  readonly username?: string;
+  readonly password?: string;
+  readonly tls?: boolean;
+  readonly skipCertVerify?: boolean;
+  readonly udp?: boolean;
```
Comment (geekdada) [inline lib/types.ts]: Also here

**Comment by geekdada** [comment]
This feature has been released with https://github.com/geekdada/surgio/releases/tag/v1.18.0!

**Pull request merged**